scalanet/test/src/io/iohk/scalanet/peergroup/kademlia/BitVectorCodecSpec.scala

```diff
@@ -0,0 +1,78 @@
+package io.iohk.scalanet.peergroup.kademlia
+
+import java.nio.ByteBuffer
+
+import io.iohk.decco.Codec
+import org.scalacheck.Arbitrary
+import org.scalacheck.Arbitrary.arbitrary
+import org.scalactic.Equivalence
+import org.scalatest.FlatSpec
+import scodec.bits.BitVector
+
+import scala.util.Random
+
+class BitVectorCodecSpec extends FlatSpec {
+
+  import BitVectorCodec._
+  import BitVectorCodecSpec._
+  import io.iohk.decco.auto._
+
+  implicit val arbitraryBitVector = Arbitrary(Generators.genBitVector())
+
+  "BitVectorCodec" should "work" in testCodec[BitVector]
+}
+
+object BitVectorCodecSpec {
+  import io.iohk.decco.BufferInstantiator.global.HeapByteBuffer
+  import org.scalatest.EitherValues._
+  import org.scalatest.Matchers._
+  import org.scalatest.prop.GeneratorDrivenPropertyChecks._
+
+  private def encodeDecodeTest[T](implicit codec: Codec[T], a: Arbitrary[T], eq: Equivalence[T]): Unit = {
```

Comment:
isn't this already implemented in decco?

Reply:
It's not. I didn't want to introduce a new dependency (scodec) into decco just for the BitVector codec. At some point, if this codec is required more widely, it might make sense to introduce a decco-bitvector module or somesuch.